src/commands/sync.js

```diff
@@ -1,12 +1,25 @@
 /* @flow */
 
 import Command from '../command'
+import { CLIError } from '../common'
 
 export default class SyncCommand extends Command {
   name = 'sync'
   description = 'Sync upstream changes'
 
-  callback() {
-    console.log('sync', this.projectsRoot)
+  async callback() {
+    const projects = await this.getProjects()
+    if (!projects.length) {
+      throw new CLIError("You don't have any projects to sync")
+    }
+    const promises = []
+    const syncRepo = project => this.spawn('git', ['fetch', '--all'], { stdio: 'inherit', cwd: project.path })
+    projects.map((project) => {
+      promises.push(syncRepo(project))
+    })
+    await Promise.all(promises).then((results) => {
```

Comment:
Also it would be wonderful if this could be structured like
```
try {
  const results = await Promise.all(promises)
  console.log('results', results)
} catch (error) {
  throw new CLIError(`Something went wrong with syncing your repos: ${error.message}`)
}
```

Also there's a flaw in current version, catch doesn't re-throw, it returns. Returning is bad, if you want to throw a new error in catch, you have to throw it